google/cloud/internal/tracing_rest_client_test.cc

```diff
@@ -107,6 +107,64 @@ TEST(TracingRestClient, Delete) {
           SpanNamed("Read"), SpanNamed("Read")));
 }
 
+TEST(TracingRestClient, HasScope) {
```

Comment:
So I personally prefer to have more independent tests. As it stands, `Delete ⊂ HasScope`. I would suggest either

A. Consolidating the `Delete` and `HasScope` test into one.

B. Isolating only the things that have to do with the request span being active. e.g. removing fluff like 
```cc
  request.AddHeader("x-test-header-3", "value3");
```

or even validating the response and its contents.

and the final expectation would just look like:
```cc
  EXPECT_THAT(spans, UnorderedElementsAre(
                         // Request span
                         AllOf(SpanNamed("HTTP/GET"),
                               SpanHasAttributes(SpanAttribute<std::string>(
                                   "test.attribute", "test.value"))),
                         // Read span on the HttpPayload
                         SpanNamed("Read"), SpanNamed("Read")));
```

Reply:
Done.